### ISSUE TITLE
Ma1sd not working if Nginx proxy disabled

### ISSUE THREAD

**Issue by enekonieto** [spantaleev/matrix-docker-ansible-deploy#1047]
If you set `matrix_nginx_proxy_enabled: false` searching user directory fails with `An internal server error occurred. Contact your administrator with reference Transaction #1620677698179`.

Peeking Ma1sd's logs I saw the reason:
```
(May 10 22:14:58 fosil matrix-ma1sd[1903298]: [XNIO-1 task-5] INFO io.kamax.mxisd.directory.DirectoryManager - Querying HS at http://matrix-nginx-proxy:12080/_matrix/client/r0/user_directory/search
May 10 22:14:58 fosil matrix-ma1sd[1903298]: [XNIO-1 task-5] ERROR io.kamax.mxisd.http.undertow.handler.SaneHandler - Transaction #1620677698179 - Unable to query the HS: I/O error: matrix-nginx-proxy: Name does not resolve)
```

Ma1sd tries to contact `matrix-nginx-proxy` but that host doesn't exist. `/matrix/ma1sd/config/ma1sd.yaml` has:
```
dns:
    overwrite:
        homeserver:
            client:
            -   name: matrix.fosil.eu
                value: http://matrix-nginx-proxy:12080
```

Fixed adding:
```
matrix_ma1sd_dns_overwrite_enabled: true
matrix_ma1sd_dns_overwrite_homeserver_client_value: "http://matrix-synapse:8008"
```

**Comment by enekonieto** [comment]
vars.yml before adding fix:

```
matrix_domain: MY-DOMAIN
matrix_coturn_turn_static_auth_secret: "SECRET"
matrix_synapse_macaroon_secret_key: "SECRET"
matrix_postgres_connection_password: 'SECRET'
matrix_synapse_workers_enabled: false

matrix_jitsi_enabled: true
matrix_jitsi_jicofo_component_secret: SECRET
matrix_jitsi_jicofo_auth_password: SECRET
matrix_jitsi_jvb_auth_password: SECRET
matrix_jitsi_jibri_recorder_password: SECRET
matrix_jitsi_jibri_xmpp_password: SECRET
matrix_server_fqn_jitsi: "meet.{{ matrix_domain }}"

matrix_synapse_enable_registration: false
matrix_client_element_registration_enabled: false

matrix_nginx_proxy_enabled: false
matrix_ssl_retrieval_method: none

matrix_synapse_admin_enabled: true

matrix_nginx_proxy_proxy_matrix_client_api_forwarded_location_synapse_oidc_api_enabled: true

matrix_coturn_enabled: true
matrix_coturn_tls_enabled: true
matrix_coturn_tls_cert_path: /etc/letsencrypt/live/SECRET/fullchain.pem
matrix_coturn_tls_key_path: /etc/letsencrypt/live/SECRET/privkey.pem
matrix_coturn_container_additional_volumes: [{"src": "/etc/letsencrypt", "dst": "/etc/letsencrypt", "options": "ro"}]

matrix_docker_network: 'nginx'

matrix_postgres_backup_enabled: true
```

**Comment by spantaleev** [comment]
Thank you for reporting this! Should be fixed now, so you shouldn't need your custom overrides.

**Comment by tna76874** [comment]
I did upgrade my server from the playbook commit ccebbc0 to commit cb7ad41 

Then, the error mentioned by [enekonieto](https://github.com/enekonieto) occured. I upgraded several other times (setup-all) but the error did not disappear.

`Mar 30 16:58:41 myserver matrix-ma1sd[3435468]: [XNIO-1 task-1] INFO io.kamax.mxisd.directory.DirectoryManager - Querying HS at http://matrix-traefik:8008/_matrix/client/r0/user_directory/search
Mar 30 16:58:41 myserver matrix-ma1sd[3435468]: [XNIO-1 task-1] ERROR io.kamax.mxisd.http.undertow.handler.SaneHandler - Transaction #1711814321351 - Unable to query the HS: I/O error: Connect to matrix-traefik:8008 [matrix-traefik/172.31.0.2] failed: Connection refused (Connection refused)`

After adding the workaround ...

`matrix_ma1sd_dns_overwrite_enabled: true
matrix_ma1sd_dns_overwrite_homeserver_client_value: "http://matrix-synapse:8008"`

it is again possible to search for users and the error disappeared.

Can I help to track down the root of the error for my setup, or should I just live with the workaround?
